examples/images_contours_and_fields/barcode_demo.py

```diff
@@ -1,33 +1,40 @@
 """
-============
-Barcode Demo
-============
-
-This demo shows how to produce a one-dimensional image, or "bar code".
+=======
+Barcode
+=======
+This demo shows how to produce a bar code.
+
+The figure size is calculated so that the width in pixels is a multiple of the
+number of data points to prevent interpolation artifacts. Additionally, the
+``Axes`` is defined to span the whole figure and all ``Axis`` are turned off.
+
+The data itself is rendered with `~.Axes.imshow` using
+
+- ``code.reshape(1, -1)`` to turn the data into a 2D array with one row.
+- ``imshow(..., aspect='auto')`` to allow for non-square pixels.
+- ``imshow(..., interpolation='nearest')`` to prevent blurred edges. This
+  should not happen anyway because we fine-tuned the figure width in pixels,
+  but just to be safe.
 """
+
 import matplotlib.pyplot as plt
 import numpy as np
 
-# Fixing random state for reproducibility
-np.random.seed(19680801)
-
-# the bar
-x = np.random.rand(500) > 0.7
-
-barprops = dict(aspect='auto', cmap='binary', interpolation='nearest')
-
-fig = plt.figure()
 
-# a vertical barcode
-ax1 = fig.add_axes([0.1, 0.1, 0.1, 0.8])
-ax1.set_axis_off()
-ax1.imshow(x.reshape((-1, 1)), **barprops)
+code = np.array([
+    1, 0, 1, 0, 1, 1, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 1, 1,
+    0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0,
+    1, 0, 1, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0, 1, 0, 0, 1, 1, 0, 1, 1, 0, 0, 1,
+    1, 0, 0, 1, 1, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 1])
 
-# a horizontal barcode
-ax2 = fig.add_axes([0.3, 0.4, 0.6, 0.2])
-ax2.set_axis_off()
-ax2.imshow(x.reshape((1, -1)), **barprops)
+pixel_per_bar = 4
+dpi = 100
```

Comment:
So not sure this matters, but we save at 200 dpi now, don't we?

Reply:
Um, I don't think so. When did that happen?

We now oversample the thumbnails in the gallery (#17734). But that does not affect the images in the examples themselves. Maybe you are mixig this up?